trieste/bayesian_optimizer.py

```diff
@@ -327,6 +328,7 @@ def optimize(
         track_state: bool = True,
         track_path: Optional[Path | str] = None,
         fit_initial_model: bool = True,
+        fit_global_model: bool = True,
```

Comment:
I don't like the name `fit_global_model`, it's too TurBO-specific...

We could either change it to `refit_models` or `fit_model_at_each_iteration` or something better?

Alternatively to avoid inflating the number of arguments we could use a single `skip_model_training` with possible values `'none'`, `'all'`, `'init'`? 

Also, we would need to do the same for `ask_tell` perhaps?